### PR TITLE
18057: Fix adding switch options via API

### PR DESCRIPTION
**Pull request by jakubderdziak**
### Description
This change allows to create correct options in text swatch attribute via REST API or PHP. The idea is to copy data required by swatch from options (keep in mind, that in case of API or PHP we will not have full options list in $attribute->getData('option'), and $attribute->getData('swatchtext'); will always return null.

### Fixed Issues (if relevant)
1. magento/magento2#18057: Swatch Text attributes options not created correctly via api

### Manual testing scenarios
For steps to reproduce, please check bug report: https://github.com/magento/magento2/issues/18057

### Contribution checklist
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

### TODO
That change will require update for unit tests.

### CODE REVIEW

**Comment by magento-cicd2** [comment]
[CLA assistant check](https://cla.dev.magento.com/magento/magento2?pullRequest=18114) <br/>All committers have signed the CLA.

**Comment by magento-engcom-team** [comment]
Hi @jakubderdziak. Thank you for your contribution 
Here is some useful tips how you can test your changes using Magento test environment.
Add the comment under your pull request to deploy test or vanilla Magento instance:
- `@magento-engcom-team give me test instance` - deploy test instance based on PR changes
- `@magento-engcom-team give me $VERSION instance` - deploy vanilla Magento instance

For more details, please, review the [Magento Contributor Assistant](https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#contributor-assist) documentation

**Review by orlangur — CHANGES_REQUESTED** [review]
Nowadays all changes must be applied to `2.3-develop` first (see https://devdocs.magento.com/guides/v2.3/contributor-guide/contributing.html#rules). Please prepare a new pull request, this one should be `on hold` until the latter is merged.

**Comment by orlangur** [comment]
Closing due to inactivity. Feel free to reach me out anytime later if you wish to continue work on this pull request and it will be reopened.
Thanks for collaboration!

**Comment by rguedes** [comment]
With this fix, can I create a Visual Swatch? How I upload the "image" ??

**Pull request closed without merging**
